src/modules/facility-loan/repayment-schedule.builder.test.ts

```diff
@@ -0,0 +1,203 @@
+import { ENUMS } from '@ukef/constants';
+import { CALENDAR_IDENTIFIERS } from '@ukef/constants/calendar-identifiers.constant';
+import { CURRENCIES } from '@ukef/constants/currencies.constant';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { CreateFacilityLoanGenerator } from '@ukef-test/support/generator/create-facility-loan-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+
+import { RepaymentScheduleBuilder } from './repayment-schedule.builder';
+
+describe('FacilityLoanService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const facilityIdentifier = valueGenerator.facilityId();
+  const bundleIdentifier = valueGenerator.acbsBundleId();
+  const dateStringTransformations = new DateStringTransformations();
+
+  let repaymentScheduleBuilder: RepaymentScheduleBuilder;
+
+  beforeEach(() => {
+    repaymentScheduleBuilder = new RepaymentScheduleBuilder(dateStringTransformations);
+  });
+
+  describe('createRepaymentSchedulesForLoan', () => {
+    const {
+      requestBodyToCreateFacilityLoanGbp,
+      requestBodyToCreateFacilityLoanNonGbp,
+      bondRepaymentSchedulesGbp,
+      ewcsRepaymentSchedulesGbp,
+      gefRepaymentSchedulesGbp,
+    } = new CreateFacilityLoanGenerator(valueGenerator, dateStringTransformations).generate({
+      numberToGenerate: 1,
+      facilityIdentifier,
+      bundleIdentifier,
+    });
+    const [newLoanGbp] = requestBodyToCreateFacilityLoanGbp;
+    const [newLoanNonGbp] = requestBodyToCreateFacilityLoanNonGbp;
+
+    describe('generates correct repayment schedules', () => {
+      it(`productTypeGroup is 'BOND' and currency is 'GBP`, () => {
```

Comment:
Nice work on these tests! Can we update the test names to describe _what_ is supposed to happen for each case as well please? E.g. maybe something like  `'generates bond repayment schedules with EU calendar identifier when product type is 'BOND' and currency is 'EUR'`?

Or if that makes the naming too difficult / long, maybe we split it with describe blocks like:

```
describe('generates bond repayment schedules when product type is 'BOND'`, () => {
   it(`with UK calendar identifier when currency is 'GBP', () => { /*...*/ });
   it(`with EU calendar identifier when currency is 'EUR'`, () => { /*...*/ });
   it(`with US calendar identifier when currency is 'GBP'`, () => { /*...*/ });
});
```